app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def udpate_post_counter_for_a_user
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
[OPTIONAL]

- It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#post_counter` method every time a new post has been saved to the database, by this you can make the `udpate_post_counter_for_a_user` method private which will increase the security.

```ruby
# app/models/post.rb
  after_save :udpate_post_counter_for_a_user

  private

  def udpate_post_counter_for_a_user
    author.increment!(:posts_counter)
  end
```